src/sandbox/components/Grid.svelte

```diff
@@ -13,33 +15,63 @@
   export let onkeydown;
   export let transparent = false;
   export let overflowHidden = false;
+  export let isEditMode;
 
   const thisSheet = getContext('store');
   const { nRows, nCols } = $thisSheet;
   const rowHeight = 20, columnWidth = 100;
 
   let focused = false;
   let editorSession = null;
+  let isFormula = false;
 
-  let container = null, editor = null;  // refs
+  let container = null, editor = null;
   const dispatch = createEventDispatcher();
 
+  let isNodeInserted = false;
+  let isArrowMode = false;
+  let insertRange = null;
+  let highlight = selection;
+  let arrowRow, arrowCol;
+
+  $: if (editorSession !== null) isFormula = detectFormula(editorSession)
+
+  $: isEditMode = (editor !== null);
+  $: if (!isEditMode) {
+    insertRange = null;
+    isArrowMode = false;
+  }
+  $: if (isEditMode) highlight = selection;
+
+  $: if (isArrowMode) [arrowRow, arrowCol] = highlight ? highlight : selection;
+  $: if (!isArrowMode) highlight = selection;
```

Comment:
isn't this redundant?

Reply:
Yes. This logic is simplified in upcoming commit.